BuildNotifications/ViewModel/Utils/RemoveTrackingObservableCollection.cs

```diff
@@ -26,7 +27,11 @@ public RemoveTrackingObservableCollection(TimeSpan removeDelay, IEnumerable<T> i
         {
             RemoveDelay = removeDelay;
             _list = new ObservableCollection<T>(initialValues);
-            _list.CollectionChanged += (sender, args) => { CollectionChanged?.Invoke(sender, args); };
+            _list.CollectionChanged += (sender, args) =>
+            {
+                Application.Current.Dispatcher?.Invoke(() =>
```

Comment:
This should theoretically not be necessary, as modifying the collection from a thread different than the UI Thread is generally not supported by WPF. This is why no dispatcher was previously used here and the exception that follows is a desired effect.

Reply:
> modifying the collection from a thread different than the UI Thread is generally not supported by WPF.

That's the point. The underlying collection is changed from a different thread and the RemoveTrackingObservableCollection should raise its event on the UI thread so WPF can do its thing.

Reply:
No, I don't think you would expect that. The normal ObservableCollection does not dispatch its .Add or .Remove methods to the UI thread. Neither does any other collection. 

If you make the mistake of modifying the collection from a different thread than the UI Thread and have its values binded, you should receive an exception. 

+ You introduce unnecessary overhead, when the current thread is already the UI Thread.

However, to reduce confusion maybe a check within the .Remove method is useful so that the Exception throws immediately and not after the remove delay, when the item is actually removed.

So 
```
public void Remove(T item) {
If (thisIsOnThreadOtherThanUi)
   throw Exception();
```

I don't think it's the collections responsibility to solve WPFs fundamental issues.